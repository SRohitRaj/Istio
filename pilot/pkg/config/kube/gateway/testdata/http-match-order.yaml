apiVersion: gateway.networking.k8s.io/v1beta1
kind: GatewayClass
metadata:
  name: istio
spec:
  controllerName: istio.io/gateway-controller
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: gateway
  namespace: istio-system
spec:
  addresses:
    - value: istio-ingressgateway
      type: Hostname
  gatewayClassName: istio
  listeners:
    - name: default
      hostname: "*.domain.example"
      port: 80
      protocol: HTTP
      allowedRoutes:
        namespaces:
          from: All
---
apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: match-order
  namespace: default
spec:
  parentRefs:
    - name: gateway
      namespace: istio-system
  rules:
    - backendRefs:
        - name: httpbin
          port: 80
      matches:
        - path:
            type: PathPrefix
            value: /api/store
        - path:
            type: RegularExpression
            value: ^/api/[1-9][0-9]*/.*$
    - backendRefs:
        - name: httpbin
          port: 80
      matches:
        - path:
            type: PathPrefix
            value: /
