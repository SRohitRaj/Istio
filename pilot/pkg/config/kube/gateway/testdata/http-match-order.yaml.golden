apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-semantics: gateway
    internal.istio.io/gateway-service: istio-ingressgateway.istio-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/gateway/default.istio-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-default
  namespace: istio-system
spec:
  servers:
  - hosts:
    - '*/*.domain.example'
    port:
      name: default
      number: 80
      protocol: HTTP
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/match-order.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: match-order-0-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - istio-system/gateway-istio-autogenerated-k8s-gateway-default
  hosts:
  - '*'
  http:
  - match:
    - uri:
        prefix: /api/store
    name: default.match-order.0
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
  - match:
    - uri:
        regex: ^/api/[1-9][0-9]*/.*$
    name: default.match-order.0
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
  - match:
    - uri:
        prefix: /
    name: default.match-order.1
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
---
